### PR TITLE
Add `PermissionService`

### PR DESCRIPTION
**Pull request by dnlkoch**
This initializes the `PermissionService` that includes all permission endpoints introduced in [PR54](https://github.com/terrestris/shogun/pull/54).

Please review @terrestris/devs.

### CODE REVIEW

File: src/service/PermissionService/index.ts [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+
+import UserInstancePermission from '../../model/security/UserInstancePermission';
+import GroupInstancePermission from '../../model/security/GroupInstancePermission';
+import UserClassPermission from '../../model/security/UserClassPermission';
+import GroupClassPermission from '../../model/security/GroupClassPermission';
+import PermissionCollectionType from '../../model/enum/PermissionCollectionType';
+
+export interface PermissionServiceOpts {
+  basePath: string;
+  keycloak?: Keycloak;
```
Comment (LukasLohoff) [inline src/service/PermissionService/index.ts:14]: just a question and non blocking: shouldn't keycloak be required here?

**Review by LukasLohoff — APPROVED** [review]
Very nice!

File: src/service/PermissionService/index.ts (hunk 1 above)
Comment (dnlkoch) [reply src/service/PermissionService/index.ts:14]: Yes, usually you'll always want to provide a `Keycloak` instance to the service, but generally speaking it's only needed if the backend interface requires a (Keycloak) bearer token for authentication. If the interface is accessible for anonymous users or no Keycloak is present at all, the client _might_ not be needed.

Comment (LukasLohoff) [reply src/service/PermissionService/index.ts:14]: allright, thanks for clarifying this :+1:

**Pull request merged**